js2-mode.el

```diff
@@ -12760,24 +12770,34 @@ i.e. ('name' 'value') = {name : { value: 3}}"
         (temp-object object)
         (temp t) ;temporay node
         (names prop-names))
-    (while (and temp names (js2-object-node-p temp-object))
+    (while (and temp names (or (js2-object-node-p temp-object)
+                               (js2-method-node-p temp-object)))
       (setq temp (js2-search-object temp-object (pop names)))
       (and (setq node temp)
-         (setq temp-object (js2-object-prop-node-right temp))))
+           (setq temp-object (if (js2-method-node-p temp)
```

Comment:
Since both `js2-method-node` and `js2-object-prop-node` inherit from `js2-infix-node`, this can be shortened to `(js2-infix-node-right temp)`.